### PR TITLE
 [TFLite] Enable reference int8 fully connected with output shape with dimensions > 2 

### PR DESCRIPTION
**Pull request by SaoirseARM**
Hi,

This PR enables reference int8 support for fully connected with output shape with dimensions > 2 in TensorFlow Lite.
Relevant issue which describes this is here: https://github.com/tensorflow/tensorflow/issues/53501 

This PR also contains an update to the SimpleTestQuantizedOutputShape3DInt16 to ensure that output shape with dimensions >2 is verified.

Thanks,
Saoirse

### CODE REVIEW

File: tensorflow/lite/kernels/internal/reference/integer_ops/fully_connected.h [hunk 1]
```diff
@@ -34,12 +34,13 @@ inline void FullyConnected(
   const int32_t output_activation_min = params.quantized_activation_min;
   const int32_t output_activation_max = params.quantized_activation_max;
   TFLITE_DCHECK_GE(filter_shape.DimensionsCount(), 2);
-  TFLITE_DCHECK_EQ(output_shape.DimensionsCount(), 2);
+  TFLITE_DCHECK_GE(output_shape.DimensionsCount(), 1);
```
Comment (haozha111) [inline tensorflow/lite/kernels/internal/reference/integer_ops/fully_connected.h:37]: should this be TFLITE_DCHECK_GE(output_shape.DimensionsCount(), 2) ?

**Review by fredrec — APPROVED** [review]

**Comment by abattery** [comment]
Could you add a e2e test case under the lite_v2_test.py?

**Comment by SaoirseARM** [comment]
Hi, Thanks for review. I have added end to end test to lite_v2_test.py. Please let me know if there is anything I need to add/change.

Best regards,
Saoirse

File: tensorflow/lite/kernels/internal/reference/integer_ops/fully_connected.h (hunk 1 above)
Comment (fredrec) [inline tensorflow/lite/kernels/internal/reference/integer_ops/fully_connected.h:37]: One more thing: shouldn't this be `TFLITE_DCHECK_GE(.., 2)` ?

File: tensorflow/lite/python/lite_v2_test.py [hunk 2]
```diff
@@ -2180,6 +2180,45 @@ def testKerasConvLSTM2DWithMoreThanOneDilationRate(self):
...
+    converter.inference_input_type = dtypes.int8
+    converter.inference_output_type = dtypes.int8
+    tflite_model = converter.convert()
+    self.assertTrue(tflite_model)
+
+    interpreter = Interpreter(model_content=tflite_model)
+    interpreter.allocate_tensors()
+    output_details = interpreter.get_output_details()
+
+    self.assertLen(output_details[0]['shape_signature'], 3)
```
Comment (abattery) [inline tensorflow/lite/python/lite_v2_test.py:2220]: could you add value comparisons between TF and TFLite models?

**Comment by gbaned** [comment]
Seems auto-merge is not happening but the changes are merged into master now, so we can close this. Thank you for the PR.

**Pull request closed without merging**